src/_pytest/python.py

```diff
@@ -681,22 +682,24 @@ def __init__(
             session=session,
             nodeid=nodeid,
         )
-        self.name = self.path.parent.name
+        self.name = self.path.name
 
     def setup(self) -> None:
+        init_mod = importtestmodule(self.path / "__init__.py", self.config)
```

Comment:
This was previously handled by `Module` importing its module file (the `__init__.py` file), now we need to do it ad-hoc.

As discussed in the issue, I think we should consider deprecating `setup_module`/`teardown_module` in `__init__.py` files for `Package`s, it's pretty obscure, but for now I retained full compat.

Reply:
makes me wonder if we want to consider deprecating xunit style setup methods alltogether as they are practically better served by fixtures

Reply:
I think it will be too disruptive at this point, but maybe with nose gone we can start considering this as well. Overall though I think it's not worth deprecating, I know a lot of people use them.

This particular case though (`setup_module` xunit fixtures in the `__init__.py` file being package scoped) is more dubious than the others, I'd be surprised to learn that anyone knows about it at all. So we might want to consider deprecating it specifically.